homeassistant/components/recorder/core.py

```diff
@@ -924,7 +924,7 @@ def _process_one_task_or_event_or_recover(self, task: RecorderTask | Event) -> N
             # that is pending before running the task
             if TYPE_CHECKING:
                 assert isinstance(task, RecorderTask)
-            if not task.commit_before:
+            if task.commit_before:
```

Comment:
This was a logic reversal and since all of our tests ran with a commit interval of zero it went unnoticed so I added an explicit test with a commit interval of 60